sys/xtimer/xtimer.c

```diff
@@ -246,14 +247,21 @@ int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t timeout)
 {
     xtimer_t t;
     mutex_thread_t mt = { mutex, (thread_t *)sched_active_thread, 0 };
+    int locked = mutex_trylock(mutex);
 
-    if (timeout != 0) {
-        t.callback = _mutex_timeout;
-        t.arg = (void *)((mutex_thread_t *)&mt);
-        _xtimer_set64(&t, timeout, timeout >> 32);
+    if (locked || (timeout == 0)) {
+        return (locked - 1) * ETIMEDOUT;
```

Comment:
This is an unnecessary optimization in my eyes. Make it a simple `if` branch and return `-ETIMEDOUT` if it is locked, otherwise 0.
The current code hurts readability.